### ISSUE TITLE
Error after updating from 1.9.3 -> 1.9.5: only yarn v2/v3 with `nodeLinker: node-modules` is supported at this time

### ISSUE THREAD

**Issue by dobesv** [vercel/turbo#4953]
### What version of Turborepo are you using?

1.9.5

### What package manager are you using / does the bug impact?

Yarn v2/v3 (node_modules linker only)

### What operating system are you using?

Linux

### Describe the Bug

When upgrading [1.9.3 -> 1.9.5](https://renovatebot.com/diffs/npm/turbo/1.9.3/1.9.5) it now prints an error:

```
Error: only yarn v2/v3 with `nodeLinker: node-modules` is supported at this time
```

We've been using turbo for many months now with Yarn PnP seemingly without problems, so it's surprising that it says this is not supported.

### Expected Behavior

Keeps working as well as it did in version 1.9.3

### To Reproduce

Probably need a yarn berry project that is not using the `node-modules` linker and compare the behavior between `1.9.3` and `1.9.5`

**Comment by flex-jonghyen** [comment]
It's surprising and frustrating when a feature that's supported is suddenly discontinued without warning. Could you tell me when yarn pnp will be available to its users again?

**Comment by dobesv** [comment]
> Could you tell me when yarn pnp will be available to its users again?

FWIW from what I can tell they have never "officially" supported yarn PnP.  However, it has been working fine in Yarn PnP for us for some time, so I'm not sure what the blocker is.

**Comment by NicholasLYang** [comment]
Hi, yeah it does seem that we accidentally permitted yarn PnP, when that was never the intention. We're currently investigating what precise features do not work well with PnP so we can restrict accordingly.

**Comment by psychobolt** [comment]
My yarn PnP project was working fine until this upgrade. 

> Hi, yeah it does seem that we accidentally permitted yarn PnP, when that was never the intention. We're currently investigating what precise features do not work well with PnP so we can restrict accordingly.

Seems like they are not even sure yet why it should be blocked, but did so.

**Comment by nathanhammond** [comment]
There were multiple problems in the past that prevented us from working with yarn3 PnP. Over the past six or nine months we've worked through a lot of them as a side effect of work we were doing, but "supporting PnP" wasn't the primary goal of those efforts.

So, as it stands, we actually believe we support it, but have not written the tests that would confirm that.

The issue here is that we previously did not hard error in all scenarios with yarn PnP, which was a "bug" that we "fixed."

Your reports of it working increase our confidence that we just need to confirm everything with tests, so we're working on that!

**Comment by nathanhammond** [comment]
This is released in 1.9.8.
